captum/attr/_core/shapley_value.py

```diff
@@ -456,3 +472,49 @@ def construct_feature_mask(
         total_features = current_num_features
         feature_mask = tuple(feature_mask)
         return feature_mask, total_features
+
+
+class ShapleyValues(ShapleyValueSampling):
+    """
+    A perturbation based approach to compute attribution, based on the concept
+    of Shapley Values from cooperative game theory. This method involves taking
+    each permutation of the input features and adding them one-by-one to the
+    given baseline. The output difference after adding each feature corresponds
+    to its attribution, and these difference are averaged over all possible
+    random permutations of the input features.
+
+    By default, each scalar value within
+    the input tensors are taken as a feature and added independently. Passing
+    a feature mask, allows grouping features to be added together. This can
+    be used in cases such as images, where an entire segment or region
+    can be grouped together, measuring the importance of the segment
+    (feature group). Each input scalar in the group will be given the same
+    attribution value equal to the change in output as a result of adding back
+    the entire feature group.
+
+    More details regarding Shapley Values can be found in these papers:
+    https://www.sciencedirect.com/science/article/pii/S0305054808000804
+    https://pdfs.semanticscholar.org/7715/bb1070691455d1fcfc6346ff458dbca77b2c.pdf
+
+    NOTE: The method implemented here is very computationally intensive, and
+    should only be used with a very small number of features (e.g. < 7).
+    This implementation simply extends ShapleyValueSampling and
+    evaluates all permutations, leading to a total of n * n! evaluations for n
+    features. Shapley values can alternatively be computed with only 2^n
+    evaluations, and we plan to add this approach in the future.
```

Comment:
It would be good to check the size of the input tensor and warn the user using it or suggest to use the sampling one ... 

This will require overriding the attribute method and making a simple check ....

Reply:
Good idea, adding!